src/Table.php

```diff
@@ -194,29 +194,27 @@ public function addColumn($name, $columnDecorator = null, $field = null)
             $field = ['type' => $field];
         }
 
-        if (is_string($name) && $name) {
-            $existingField = $this->model->hasField($name);
-        } else {
-            $existingField = null;
-        }
-
-        if ($existingField === null) {
+        if ($name === null) {
             // table column without respective field in model
             $field = null;
-        } elseif (!$existingField) {
+        } elseif (!$this->model->hasField($name)) {
             // Add missing field
             $field = $this->model->addField($name, $field ?: []);
 
             $field->never_persist = true;
-        } elseif (is_array($field)) {
-            // Add properties to existing field
-            $existingField->setDefaults($field);
-            $field = $existingField;
-        } elseif (is_object($field)) {
-            throw (new Exception('Duplicate field'))
-                ->addMoreInfo('name', $name);
         } else {
-            $field = $existingField;
+            $existingField = $this->model->getField($name);
+
+            if (is_array($field)) {
+                // Add properties to existing field
+                $existingField->setDefaults($field);
```

Comment:
How about
```
$field = $existingField->setDefaults($field);
```

Reply:
This was not intended to be fixed by this PR... But I simplified it...